### PR TITLE
Update CHANGELOG for v0.10.0 release

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,10 +7,135 @@ and this project adheres to [Semantic Versioning].
 
 ## [0.10.0] — unreleased
 
+The highlights of this release are support for [Shanghai] execution specification upgrade
+and implementation of EOF "v1.0". There are also big enhancements to the EVM testing tools
```
Comment (axic) [inline CHANGELOG.md:11]: I'd highlight t8n here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,10 +7,135 @@ and this project adheres to [Semantic Versioning].
...
+and infrastructure.
+
+Traditionally, the EVM performance has been improved as well.
```
Comment (axic) [inline CHANGELOG.md:14]:
```suggestion
As is tradition, the EVM performance has been improved as well.
```

**Review by axic — APPROVED** [review]
I had no patience to check if any PR was missed, but on the outset it looks good.

Preferably wait for @gumb0 or @rodiazet's review too.

**Comment by chfast** [comment]
> I had no patience to check if any PR was missed, but on the outset it looks good.

Some less important PRs were omitted.

**Review by yperbasis — APPROVED** [review]

**Review by gumb0 — APPROVED** [review]

**Pull request merged**